addons/purchase/report/purchase_report.py

```diff
@@ -61,80 +64,84 @@ def _select(self):
-                    avg(100.0 * (l.price_unit / COALESCE(NULLIF(cr.rate, 0),1.0) * l.product_qty) / NULLIF(ip.value_float*l.product_qty/u.factor*u2.factor, 0.0))::decimal(16,2) as negociation,
-                    sum(ip.value_float*l.product_qty/u.factor*u2.factor)::decimal(16,2) as price_standard,
-                    (sum(l.product_qty * l.price_unit / COALESCE(NULLIF(cr.rate, 0), 1.0))/NULLIF(sum(l.product_qty/u.factor*u2.factor),0.0))::decimal(16,2) as price_average,
+                    sum(l.price_total / COALESCE(po.currency_rate, 1.0))::decimal(16,2) as price_total,
+                    (sum(l.product_qty * l.price_unit / COALESCE(po.currency_rate, 1.0))/NULLIF(sum(l.product_qty/line_uom.factor*product_uom.factor),0.0))::decimal(16,2) as price_average,
                     partner.country_id as country_id,
                     partner.commercial_partner_id as commercial_partner_id,
                     analytic_account.id as account_analytic_id,
-                    sum(p.weight * l.product_qty/u.factor*u2.factor) as weight,
-                    sum(p.volume * l.product_qty/u.factor*u2.factor) as volume
+                    sum(p.weight * l.product_qty/line_uom.factor*product_uom.factor) as weight,
+                    sum(p.volume * l.product_qty/line_uom.factor*product_uom.factor) as volume,
+                    sum(l.price_subtotal / COALESCE(po.currency_rate, 1.0))::decimal(16,2) as untaxed_total,
+                    sum(l.product_qty / line_uom.factor * product_uom.factor) as qty_ordered,
+                    sum(l.qty_received / line_uom.factor * product_uom.factor) as qty_received,
+                    sum(l.qty_invoiced / line_uom.factor * product_uom.factor) as qty_billed,
+                    sum(l.product_qty / line_uom.factor * product_uom.factor) - sum(l.qty_received / line_uom.factor * product_uom.factor) as qty_to_be_billed
         """ % self.env['res.currency']._select_companies_rates()
         return select_str
 
     def _from(self):
         from_str = """
             purchase_order_line l
-                join purchase_order s on (l.order_id=s.id)
-                join res_partner partner on s.partner_id = partner.id
+                join purchase_order po on (l.order_id=po.id)
+                join res_partner partner on po.partner_id = partner.id
                     left join product_product p on (l.product_id=p.id)
                         left join product_template t on (p.product_tmpl_id=t.id)
-                        LEFT JOIN ir_property ip ON (ip.name='standard_price' AND ip.res_id=CONCAT('product.product,',p.id) AND ip.company_id=s.company_id)
-                left join uom_uom u on (u.id=l.product_uom)
-                left join uom_uom u2 on (u2.id=t.uom_id)
+                        LEFT JOIN ir_property ip ON (ip.name='standard_price' AND ip.res_id=CONCAT('product.product,',p.id) AND ip.company_id=po.company_id)
```

Comment:
you don't need ir_property ip anymore it seems

Reply:
We have different price per company .. that's why we need to fetch it from ir_properties

Reply:
Okay but it seems not used at all in the SQL query.

Reply:
@amoyaux  ir_property ip removed and also changed help on currency_rate.